crates/rpc-types-engine/src/payload/mod.rs

```diff
@@ -19,3 +22,53 @@ pub enum OpExecutionPayload {
     /// V4 payload
     V4(OpExecutionPayloadV4),
 }
+
+impl OpExecutionPayload {
+    /// Returns a reference to the V2 payload, if any.
+    pub const fn as_v2(&self) -> Option<&ExecutionPayloadV2> {
+        match self {
+            Self::V2(payload) => Some(payload),
+            Self::V3(payload) => Some(&payload.payload_inner),
+            Self::V4(payload) => todo!(),
+        }
+    }
+
+    /// Returns a mutable reference to the V2 payload, if any.
+    pub fn as_v2_mut(&mut self) -> Option<&mut ExecutionPayloadV2> {
```

Comment:
```suggestion
    pub fn as_v2_mut(&mut self) -> &mut ExecutionPayloadV2 {
```
same here, we always have v2